### DISCUSSION TITLE
Laser profile for donut like Laguerre Gauss Beams

### DISCUSSION THREAD

**Discussion by VinithSamson11** [SmileiPIC/Smilei#697] [Q&A] [answered]
Hi,

I have a doubt regarding the generic laser profile function. The generic laser block for AM geometry is given by,

Laser(
    box_side = "xmin",
    space_time_profile_AM = [ Br_mode0, Bt_mode0, Br_mode1, Bt_mode1, ... ]
)

I have a scalar field of donut like Laguerre Gauss beam "B(x, r, theta, t)", how can I get the Br_mode0, Bt_mode0, Br_mode1, Bt_mode1....etc., from the provided scalar field "B(x, r, theta, t)" ?

**Comment by beck-llr** [comment, marked as answer]
Hi
 please read https://smileipic.github.io/Smilei/Understand/azimuthal_modes_decomposition.html.
The answer to your question is found just below equation 42.
Let us know if that helped.

**Comment by VinithSamson11** [comment]
Yes, it helped. Thank You so much.

**Comment by VinithSamson11** [comment]
Hi @beck-llr ,

I also have some confusions regarding the `space_time_profile` in the `Laser()` block for "3Dcartesian" geometry.

The syntax is given by,
`Laser(
           box_side = "xmin",
           space_time_profile = [ By_profile, Bz_profile ]
)`

The mathematical Bz profile function which I am using is a 4D function, that has dependencies on "x, y, z and t". But when I am defining a python function `Bz(x, y, z, t)` , I am getting the following warning & error:

`[WARNING](0) src/Profiles/Profile.cpp:191 (Profile) Profile 'Laser[0].space_time_profile[1]' has 4 arguments but requires 3`
`On rank 0 [Python] TypeError: Bz() missing 1 required positional argument: 't'`
`[ERROR](0) src/Profiles/Profile.cpp:269 (Profile) Profile 'Laser[0].space_time_profile[1]': does not seem to return a correct value`

Please let me know how can I define a 4D laser `space_time_profile` python function in 3Dcartesian geometry.

**Comment by Z10Frank** [comment]
Also, in the `AMcylindrical` geometry, the `space_time_profile_AM` you need to provide for each mode are functions of `(r,t)` and not `x,r,theta,t`. At the `box_side = "xmin"` the `x` is equal to 0, and the geometry will take care of the dependence on theta according to the mode number.

In 3D for a laser entering from `box_side = "xmin"` you would have a dependence on `y,z,t`. An example of Laguerre-Gauss laser injected in 3D is found in the namelist of this tutorial: https://smileipic.github.io/tutorials/advanced_vtk.html

You need to do some trigonometry to adapt the laser analytical expression to the mode definitions  [in this part of the documentation](https://smileipic.github.io/Smilei/Understand/azimuthal_modes_decomposition.html#decomposition-of-vector-fields) and pointed by @beck-llr .

**Comment by VinithSamson11** [comment]
Hi @Z10Frank,

Yes, I understood. At the box boundary, `x=0` is fine, but when the profile propagates in x direction, the profile should have a dependence of `x` (according to the analytical LG profile expression in [wikipedia](https://en.wikipedia.org/wiki/Gaussian_beam#Laguerre-Gaussian_modes) ). So my question is whether SMILEI is taking care of this `x` dependence while running the simulation or its assigning a default value for `x` coordinate in the profile expression throughout the simulation?  

I am currently using the namelist of the tutorial: https://smileipic.github.io/tutorials/export_VTK_namelist.py as a reference. Can I get the analytical expression of the `Bz` field used in the above namelist ? It would be better to know the analytic expression used, to understand it clearly.  

Thank You.

**Comment by Z10Frank** [comment]
I agree that if a laser pulse was initialised entirely in the window from t=0 you should initialise also the x dependence according to the analytical expression, but for a Laser initialised from a border, the Maxwell's equations solver will ensure that this boundary condition will make these field enter the window following Maxwell's equations. The laser pulse, once inside the window, will respect the whole analytical expression (approximately) if your laser parameter are within the limits of the chosen approximation to derive the analytical expression (paraxial approximation and pulse much longer than the laser wavelength).

The python functions for the Laguerre-Gauss beam in 3D and the Gaussian beam in cylindrical coordinates in that namelist should follow the same analytical expressions in your page. The only differences are that they are computed at x=0, multiplied by the carrier wave at angular frequency `omega` and multiplied by a time envelope to have a pulse of finite length. The vector `focus` contains the coordinates of the focal point.

**Reply by VinithSamson11** [reply]
Thank You so much for the explanation :)

**Reply by Z10Frank** [reply]
Hello, I implemented the initialisation of a Laguerre-Gauss mode in the `AMcylindrical` simulation of this tutorial:
https://smileipic.github.io/tutorials/advanced_vtk.html

**Comment by hafizmdyasir** [comment]
Dear all,
I copied the tutorial given on the [VTK tutorial page](https://smileipic.github.io/tutorials/advanced_vtk.html). I then used ParaView to visualize the intensity. As mentioned on the tutorial page `This type of laser has a wavefront that looks like a corkscrew and an intensity distribution that looks like a double corkscrew.`. The image is shown here. I hope the structure is visible.
<img width="1112" height="718" alt="Intensity_Zoom_300" src="https://github.com/user-attachments/assets/1da9efa2-33db-4a14-91e2-56b921248424" />


What I don't understand is why the double corkscrew. If l=1, shouldn't I get a single helix? If this is a simulation artifact, does anyone have any scientific references to the same.

Thanks in advance

**Reply by Z10Frank** [reply]
A given positive value of the transverse electric field belongs to a surface with same phase that looks like a corkscrew. 
The same can be said for a field value with opposite sign and same magnitude, but the surface will be another one.

The intensity, which is dominated by the square of the transverse field, will be always positive, also in the zones where the field is negative, so you will observe a double corkscrew.

If you visualise `E_\perp` limiting the colormap e.g. from 0.1 to 1.0, you will se a corkscrew that shows only the positive values in that interval. If you visualise the square of that quantity, you will see a double corkscrew, corresponding to the parts where `E_perp` is positive and negative, with the square of its absolute value in the selected interval.

**Reply by hafizmdyasir** [reply]
Thank you for the prompt response.

This was my initial thought as well and it makes sense to me. However, I saw a couple of thesis(es?) on the web where the intensity profile is shown to be a single helix for l=1 (see https://core.ac.uk/download/pdf/18423971.pdf page 27). To quote them verbatim, 

>  For a given value of l, 
the LG beam has l intertwined helical phase structures (helicoidal twists along a 
wavelength) possessing a screw singularity on the beam axis (Figure 1.1b). As 
metaphorically explained by Padgett: ‘if there is a single helix, the wavefront looks 
like a screw thread; if there are two helices, the wavefront looks like DNA; and for 
three helices, the wavefronts have the shape of fusilli pasta.

Is that calculated differently or something?

**Reply by Z10Frank** [reply]
I don't see any contradiction, in Fig. 1.1b the helices are showed for the wavefront, which is referred to the field, not the intensity. If in Paraview you use volume rendering on the field, limiting the colormap to a certain positive interval, a LG mode with l=1 will show a single helix. If you use the same interval but you plot the intensity, the helix will become double. For l=2, the helix of the field is double, and in the intensity you would see a quadruple helix etc. 

For the LG definition, just check the definition in the namelist and in the thesis. Normally the various sources have equivalent definitions except for an amplitude factor, or the choice of "sinusoidal" or "helical" LG mode definition.

**Comment by hafizmdyasir** [comment]
Hi sir @Z10Frank 
I've been looking into the comparison of LG and gaussian modes. I integrated the squared field distribution over the transverse plane to calculate a factor that makes total energy across the transverse plane in both modes the same. I obtained a factor `a_0(LG) = a_0(gauss) * sqrt(p! / (p+|l|)!)`. However, multiplying this with `a_0` and then calculating `Utot` from a simulation (with no species present, of course) gives me different results. The gaussian mode gives almost twice the value than the LG mode. 

Am i making a mistake in calculating the factor? I used the following field distribution:
1. `E_g = a_0 (w_0/w) . exp(-r^2/w^2) . phase terms`
2. `E_LG = a_0 (w_0/w) . (r*√2/w)^|l| . exp(-r^2/w^2) . L_pl(2r^2/w^2) . phase terms`

When calculating |E|^2, the phase terms cancel. Are these field distributions correct?

**Reply by Z10Frank** [reply]
You can find the coefficients that normalise the power along one plane, transverse to the propagation axis, in the Laguerre-Gauss section here https://en.wikipedia.org/wiki/Gaussian_beam .

You are computing is the total power along the surface of a transverse plane, to have the energy you need to include a temporal envelope and take it into account in the calculation.

I suggest to check also the normalisations.
